src/network/axiosClient.js

```diff
@@ -0,0 +1,23 @@
+import axios from "axios";
+import { toast } from "react-toastify";
+
+const axiosClient = axios.create();
+
+axiosClient.defaults.baseURL = `${import.meta.env.VITE_BASE_URL}`;
+
+axiosClient.interceptors.response.use( response => {
+  
+  return response;
+}, (error) => {
+
+  if (error.response.status === 500) {
+    toast.error("Something went wrong, please try again!");
+  } 
+  return Promise.reject(error);
+});
+
+export const getRequest = async url => {
+  const users = await axiosClient.get(`${url}`)
+  return users.data
```

Comment:
This is generic function for get request
```suggestion
  const response = await axiosClient.get(`${url}`)
  return users.data
```

or 

```suggestion
  return await axiosClient.get(`${url}`).data
```